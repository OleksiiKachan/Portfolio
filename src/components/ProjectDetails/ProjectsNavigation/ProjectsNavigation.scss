@import '../../../assets/style/variables';

.projectsNavigation {
  padding: 1rem 2rem;
  &__row {
    display: flex;
    align-items: center;
    justify-content: space-between;
    &_align-right {
      justify-content: flex-end;
    }
  }

  &__link {
    display: flex;
    align-items: center;
    text-decoration: none;
    color: $color-link;
    padding-bottom: 1rem;
  }

  &__linkArrow {
    margin: 0 1rem;
  }

  &__linkCaption {
    font-weight: 700;
  }
}

@media screen and (max-width: $breakpoint-tablet) {
  .projectsNavigation {
    padding: 1rem 1rem;

    &__linkArrow {
      margin-left: 0;
    }
  }
}
