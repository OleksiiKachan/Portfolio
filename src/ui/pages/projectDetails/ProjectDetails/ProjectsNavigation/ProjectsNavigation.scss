@import '../../../../assets/style/variables';
@import '../../../../assets/style/mixins';

.projectsNavigation {
  padding: 1rem 2rem;

  @include respond(tablet) {
    padding: 1rem 1rem;
  }

  &__row {
    display: flex;
    align-items: center;
    justify-content: space-between;
    &_align-right {
      justify-content: flex-end;
    }
  }

  &__link {
    display: flex;
    align-items: center;
    text-decoration: none;
    color: $color-link;
    padding-bottom: 1rem;
  }

  &__linkArrow {
    margin: 0 1rem;

    @include respond(tablet) {
      margin-left: 0;
    }
  }

  &__linkCaption {
    font-weight: 700;
  }
}
