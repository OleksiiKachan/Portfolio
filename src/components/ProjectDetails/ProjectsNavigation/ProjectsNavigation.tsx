import React from 'react';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import './ProjectsNavigation.scss';

const icon_arrowBack = require('../../../assets/icons/icon_arrowBack-link.svg');

/*
    <ProjectsNavigation
      className=''
    />
*/

type PropsType = {
  className?: string;
};

export default ({ className }: PropsType) => {
  return (
    <div className={classNames('projectsNavigation', className)}>
      <div className={classNames('projectsNavigation__row')}>
        <Link
          to="/projects"
          className={classNames(
            'projectsNavigation__link',
            'projectsNavigation__link_align-left'
          )}
        >
          <img
            src={icon_arrowBack}
            alt="Left looking arrow"
            className={classNames('projectsNavigation__linkArrow')}
          />
          <span className={classNames('projectsNavigation__linkCaption')}>
            Back to Projects List
          </span>
        </Link>
      </div>
    </div>
  );
};
