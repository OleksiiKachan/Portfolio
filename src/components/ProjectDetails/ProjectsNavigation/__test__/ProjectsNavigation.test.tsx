import React from 'react';
import { shallow } from 'enzyme';
import ProjectsNavigation from '..';

describe('ProjectsNavigation', () => {
  it('Snapshot', () => {
    const component = shallow(<ProjectsNavigation />);
    expect(component).toMatchSnapshot();
  });
});
