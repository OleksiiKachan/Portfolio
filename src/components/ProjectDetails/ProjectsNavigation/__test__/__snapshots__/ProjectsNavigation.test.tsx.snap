// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsNavigation Snapshot 1`] = `
<div
  className="projectsNavigation"
>
  <div
    className="projectsNavigation__row"
  >
    <Link
      className="projectsNavigation__link projectsNavigation__link_align-left"
      to="/projects"
    >
      <img
        alt="Left looking arrow"
        className="projectsNavigation__linkArrow"
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "icon_arrowBack-link.svg",
          }
        }
      />
      <span
        className="projectsNavigation__linkCaption"
      >
        Back to Projects List
      </span>
    </Link>
  </div>
</div>
`;
